CRM/Contribute/BAO/Contribution.php

```diff
@@ -4721,15 +4721,28 @@ public static function sendMail(&$input, &$ids, $contributionID, &$values, $recu
       throw new CRM_Core_Exception('Contribution does not exist');
     }
     $contribution->loadRelatedObjects($input, $ids, TRUE);
-    // set receipt from e-mail and name in value
     if (!$returnMessageText) {
-      $session = CRM_Core_Session::singleton();
-      $userID = $session->get('userID');
-      if (!empty($userID)) {
-        list($userName, $userEmail) = CRM_Contact_BAO_Contact_Location::getEmailDetails($userID);
-        $values['receipt_from_email'] = CRM_Utils_Array::value('receipt_from_email', $input, $userEmail);
-        $values['receipt_from_name'] = CRM_Utils_Array::value('receipt_from_name', $input, $userName);
+      // Set receipt from e-mail and name in value if we are sending the message.
+      $userEmail = NULL;
+      $userName = NULL;
+
+      // First, try to retrieve default name and email address.
+      $domainEmailValues = CRM_Core_BAO_Domain::getNameAndEmail();
+      if (!empty($domainEmailValues)) {
+        $userName = $domainEmailValues[0];
+        $userEmail = $domainEmailValues[1];
+      }
+
+      if (empty($userEmail)) {
+        // If that didn't work, try to use the logged in user.
+        $session = CRM_Core_Session::singleton();
```

Comment:
@jmcclelland Jamie this should probably be changed to $userID  = CRM_Core_Session::getLoggedInContactID()